extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/administrative/MongoInstall.java

```diff
@@ -0,0 +1,210 @@
+                .setEnabled(installConfig.isPcjIndexEnabled());
+        if(installConfig.getFluoPcjAppName().isPresent()) {
+            final String fluoPcjAppName = installConfig.getFluoPcjAppName().get();
+            pcjDetailsBuilder.setFluoDetails(new FluoDetails( fluoPcjAppName ));
+        }
+
+        final RyaDetails details = RyaDetails.builder()
+                // General Metadata
+                .setRyaInstanceName(instanceName)
+                .setRyaVersion( getVersion() )
+
+                // Secondary Index Values
+                .setGeoIndexDetails(
+                        new GeoIndexDetails(installConfig.isGeoIndexEnabled()))
+                .setTemporalIndexDetails(
+                        new TemporalIndexDetails(installConfig.isTemporalIndexEnabled()))
+                .setFreeTextDetails(
+                        new FreeTextIndexDetails(installConfig.isFreeTextIndexEnabled()))
+                .setEntityCentricIndexDetails(
+                        new EntityCentricIndexDetails(installConfig.isEntityCentrixIndexEnabled()))
+                .setPCJIndexDetails(  pcjDetailsBuilder.build() )
+
+                // Statistics values.
+                .setProspectorDetails(
+                        new ProspectorDetails(Optional.<Date>absent()) )
+                .setJoinSelectivityDetails(
+                        new JoinSelectivityDetails(Optional.<Date>absent()) )
+                .build();
+
+        // Initialize the table.
+        detailsRepo.initialize(details);
+
+        return details;
+    }
+
+    /**
+     * Builds a {@link MongoRdfConfiguration} object that will be used by the
+     * Rya DAO to initialize all of the tables it will need.
+     *
+     * @param connectionDetails - Indicates how to connect to Mongo. (not null)
+     * @param details - Indicates what needs to be installed. (not null)
+     * @return A Rya Configuration object that can be used to perform the install.
+     */
+    private static MongoDBRdfConfiguration makeRyaConfig(final MongoConnectionDetails connectionDetails, final RyaDetails details) {
+        final MongoDBRdfConfiguration conf = new MongoDBRdfConfiguration();
+
+        // The Rya Instance Name is used as a prefix for the index tables in Mongo.
+        conf.set(RdfCloudTripleStoreConfiguration.CONF_TBL_PREFIX, details.getRyaInstanceName());
+
+        // Enable the indexers that the instance is configured to use.
+        conf.set(ConfigUtils.USE_PCJ, "" + details.getPCJIndexDetails().isEnabled() );
+        conf.set(ConfigUtils.USE_GEO, "" + details.getGeoIndexDetails().isEnabled() );
+        conf.set(ConfigUtils.USE_FREETEXT, "" + details.getFreeTextIndexDetails().isEnabled() );
+        conf.set(ConfigUtils.USE_TEMPORAL, "" + details.getTemporalIndexDetails().isEnabled() );
+        conf.set(ConfigUtils.USE_ENTITY, "" + details.getEntityCentricIndexDetails().isEnabled());
+
+        // XXX The Mongo implementation of the secondary indices make need all
```

Comment:
Do you actually need this on the Mongo side? This looks like copy paste from accumulo.